lib/FrameManager.js

```diff
@@ -283,13 +283,29 @@ class Frame {
    * @return {!Promise<?ElementHandle>}
    */
   async $(selector) {
+    const document = await this._document();
+    const value = await document.$(selector);
+    await document.dispose();
+    return value;
+  }
+
+  /**
+   * @return {!Promise<!ElementHandle>}
+   */
+  async _document() {
```

Comment:
don't you want to cache this? Otherwise, everything becomes two-hops instead of a one hop.

Reply:
Done.